docs/custom-stack.asciidoc

```diff
@@ -271,10 +271,10 @@ use one of the functions below:
 * <<apm-set-tag,`apm.setTag()`>> - Call this to enrich collected performance data and errors with simple key/value strings that you think will help you debug performance issues and errors (tags are indexed in Elasticsearch)
 
 [float]
-[[custom-stack-compatibility]]
```

Comment:
Looking through similar past issues across the product, I think we should update all of the anchor tags to `[[supported-technologies]]`. It keeps the documentation cleaner and the redirect will ensure old links are still working. 

```suggestion
[[custom-stack-compatibility]]
```

Reply:
Looking through the changes you made now, I think I misunderstood what you were asking. What do you mean by "the same make"

Reply:
Today we have a few pages with a section that contains a link to `compatibility.html`. This link should obviously get updated as part of this PR.

The section it self have a headline and that headline have a hardcoded anchor tag to make sure that even if we change the text in the headline, the anchor tag stays the same.

That way deep links to this section will keep working. In this particular case the anchor tag have the word "compatibility" in it, but even though we're changing the headline to "Supported Technologies" we technically don't have to also chance the anchor tag.

Here's an example of one such section: https://www.elastic.co/guide/en/apm/agent/nodejs/current/custom-stack.html#custom-stack-compatibility

Reply:
Really appreciate the explanation @watson. So it's my understand then that there are three cases to be aware of if/when updating anchors:

1. **External Links** (non elastic links) - These will be handled by the redirect.
2. **Internal APM links** (anchors) - These are a simple search and replace.
3. **Internal Docs links** (on non-APM repos) - This is what would actually be broken. Unless we searched all elastic docs repos and created PRs to change the anchor links. 

Sound right?

Reply:
We can fix 2 and 3 our selfs, but unfortunately not number 1 as we can't redirect anchor tags (unless we have some magic JavaScript on the page to hijack it which we didn't last I checked). 

Anchor tags are purely handled by the users browser - so our servers never see them and can therefore not redirect them.

So if we want external links to keep working, we have to keep the original anchor tag.

Reply:
Sorry, one more clarifying question. So for number 1: 

In [this discuss post](https://discuss.elastic.co/t/agent-cant-track-oracle-sql-requests/146581/2) you link the user to `/compatibility.html#compatibility-tracing-and-instrumentation`

If we were to change the anchor and set up a redirect, here's what would happen:

The redirect we set up will forward the user from `/compatibility.html` to `/supported-technologies.html`, but the anchor tag of `#compatibility-tracing-and-instrumentation` doesn't exist. So the user is sent to  `/supported-technologies.html#compatibility-tracing-and-instrumentation` which just defaults to the top of the page since the anchor doesn't exist?

Reply:
Correct. That's why I was suggesting that we keep the names of the anchor tags and only change the name of html page.

The only downside to this is estetics, as the anchor tags will carry the name of the old page. But from a functional standpoint they will work just fine.

Reply:
Got it! Thanks for the explanations.

I agree then, change the name of the page, keep the names of the anchor tags.

Reply:
Yes, I think per the conversation Watson and I had, if we change the anchors inside the `supported-technologies` doc, there are a number of external links that will be impacted. Seems like it's better to leave those anchors as-is for now (e.g. keep `[[compatibility-frameworks]]`).

Also, line 34 of `intro.asciidoc` needs to be updated.

Reply:
Also, is there a reason that the custom, express, hapi, koa, and restify pages all have a section linking to compatibility/supported-tech, but the lambda page doesn't?

Reply:
No I think it's an oversight on our part that lambda didn't have these. I added the prefix originally because I wasn't sure if these anchor tags had to be globally unique. I'm still not sure if it matters?

Reply:
Anchor tags only have to be unique within the individual book. So as long as they're unique to the node.js docs, we're fine.

Reply:
Ah ok, well then it makes sense to add them to the lambda docs as well I think. We'll most likely make similar docs for Azure and Google cloud